code/controllers/subsystem/augury.dm

```diff
@@ -18,40 +18,37 @@ SUBSYSTEM_DEF(augury)
 	doombringers -= A
 
 /datum/controller/subsystem/augury/fire()
-	var/biggest_doom = null
-	var/biggest_threat = null
+	var/biggest_doom
+	var/biggest_threat
 
-	for(var/db in doombringers)
-		var/datum/d = db
-		if(!d || QDELETED(d))
+	for(var/datum/d as() in doombringers)
+		if(QDELETED(d))
 			doombringers -= d
 			continue
 		var/threat = doombringers[d]
 		if((biggest_threat == null) || (biggest_threat < threat))
 			biggest_doom = d
 			biggest_threat = threat
 
-	if(doombringers.len)
-		for(var/i in GLOB.player_list)
-			if(isobserver(i) && (!(observers_given_action[i])))
+	if(length(doombringers))
```

Comment:
Isn't looping indexes more performant than looping observers then checking if they're in the list?

Reply:
Considering I'm nuking this entire subsystem as soon as I get my open PR backlog cleared, I wouldn't worry about it.